code/modules/projectiles/guns/energy/cyborg.dm

```diff
@@ -0,0 +1,126 @@
+/obj/item/gun/energy/robotic/proc/gun_flag_check(var/flag_to_check) //Checks for the flag of the gun.
+	return (borg_flags & flag_to_check)
+
+/// The base gun types. Build off these four.
+/obj/item/gun/energy/robotic
+	name = "Cybernetic Gun"
+	desc = "A gun commonly used by cyborgs and other robotic lifeforms to stun"
+	var/borg_flags = COUNTS_AS_ROBOT_GUN //We add flags to this!
+	self_recharge = 1
+	use_external_power = 1
+	projectile_type = /obj/item/projectile/beam/lasertag //This is the base gun and should never be used.
+
+/obj/item/gun/energy/robotic/laser
+	borg_flags = COUNTS_AS_ROBOT_GUN | COUNTS_AS_ROBOT_LASER
+	projectile_type = /obj/item/projectile/beam
+
+/obj/item/gun/energy/robotic/taser
+	name = "Cybernetic Taser"
+	desc = "An integrated taser that is used to neutralize foes."
+	borg_flags = COUNTS_AS_ROBOT_GUN | COUNTS_AS_ROBOT_TASER
+	icon_state = "taser"
+	item_state = null	//so the human update icon uses the icon_state instead.
+	projectile_type = /obj/item/projectile/beam/stun
+	charge_cost = 400
+	recharge_time = 7 //Time it takes for shots to recharge (in ticks)
+
+/obj/item/gun/energy/robotic/disabler
```

Comment:
should be removed from the dog_modules file as it's now here and the original no longer in use

Reply:
ok ye that might be an issue... I have no clue if anyone uses those elsewhere